### ISSUE TITLE
[Bug] Change RequiredScope to be based on policies

### ISSUE THREAD

**Issue by plamber** [AzureAD/microsoft-identity-web#1002]
**Which version of Microsoft Identity Web are you using?**
1.6.0

**Where is the issue?**
* Web API
    * [ ] Protected web APIs (validating tokens)
    * [x] Protected web APIs (validating scopes)
    * [ ] Protected web APIs call downstream web APIs

**Repro**
In your sample Function app application you are using the "RequiredScope" attribute to limit the access to the Function app. This does not work in this context [codesnippet](https://github.com/AzureAD/microsoft-identity-web/blob/e81df7cff189a160e69192b31a0d406333d51ae6/ProjectTemplates/templates/Functions-CSharp/SampleFunc.cs#L33-L35)

To my understanding this is due to the different way Azure Functions follow the authorization life-cycle.

You can limit the access by using the deprecated "VerifyUserHasAnyAcceptedScope(ScopesRequired);"
Here an example on how to limit the required scope.

```
[FunctionName("token")]
[RequiredScope("Does not work")]
public async Task<IActionResult> Run(
    [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = null)] HttpRequest req)
{
    var (authenticationStatus, authenticationResponse) =
        await req.HttpContext.AuthenticateAzureFunctionAsync();

    await req.HttpContext.AuthenticateAzureFunctionAsync();
    if (!authenticationStatus) return authenticationResponse;

    // this works
    req.HttpContext.VerifyUserHasAnyAcceptedScope(ScopesRequired);

    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();

    return new OkObjectResult(requestBody);
}
```

**Expected behavior**
- Required scope should work with Azure Functions

**Actual behavior**
RequiredScope is ignored. The only way is to use VerifyUserHasAnyAcceptedScope

**Possible solution**
Most likely you have to create a dedicated class for the Azure Functions. You can find an example how to implement this [here](https://medium.com/@psaillesh/do-you-know-azure-function-have-function-filters-1df3bfbb9968). The challenge is that this feature is in preview [a while](https://github.com/Azure/azure-webjobs-sdk/issues/1284).

Alternatively, provide a dedicated VerifyUserHasAnyAcceptedScope function intended for Azure Functions.

**Comment by jmprieur** [comment]
Thanks for the heads-up @plamber

**Comment by jennyf19** [comment]
@plamber we are removing the deprecated flag from  the method `VerifyUserHasAnyAcceptedScope`, so that will be in our 1.7 release. thanks.

**Comment by jennyf19** [comment]
Included in [1.18.0 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.18.0)
